lib/pr_checker.js

```diff
@@ -169,9 +169,30 @@ class PRChecker {
       logger.warn('No full CI runs detected');
     }
 
+    let lastCiDate = null;
```

Comment:
Nit: could this be `let lastCiDate;` — I don't think it strictly needs to be `null`.

Reply:
if no ci in that pr, the lastCiDate could be null.

Reply:
@buji1993 since lastCiDate is not needed to be null has no use in this case and therefore not be assigned it make code simpler.

Reply:
@buji1993 I think @apapirovski meant we can just leave it as `undefined` because it's falsy as well.

Before we sort out how to warn about different types of CI runs, I think we should at least change the logic here a bit and make it a `lastCI` which is "the oldest CI (regardless of type) that covers every commit", and then put the CI type in warning.

Reply:
@joyeecheung I think either lastCI or oldestCI is not right in this situation. As you said some CI covers others , these CIs are not individual. now I use the oldestCI in ciMap and log the oldest ci type. FYI